packages/web/src/components/edit-track/EditTrackForm.tsx

```diff
@@ -186,6 +210,52 @@ const TrackEditForm = (
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [getArtworkUrl(updatedArtwork), setArtworkValue])
 
+  const onClickReplace = useCallback(
+    async (file: File) => {
+      const processedFiles = await Promise.all(
+        processFiles([file], (name, reason) => {
+          console.error('Failed to process file', name, 'due to', reason)
+          // todo: show error state
+        })
+      )
+
+      if (processedFiles?.length > 0) {
+        const newFile = processedFiles[0]!
```

Comment:
weird that you need the `!` here

Looks like `processedFiles` actually can return `[null]` so I think we should actually `.filter(removeNullable)` before checking length. Should ensure we don't get here with an empty `newFile` value and remove the need for the `!`